src/events/client/voiceChannelLeave.ts

```diff
@@ -4,10 +4,17 @@ import { checkBotIsAlone } from "../../helpers/discord_utils";
 
 export default async function voiceChannelLeaveHandler(member: Eris.Member, oldChannel: Eris.VoiceChannel) {
     const guildID = oldChannel.guild.id;
+    const gameSession = state.gameSessions[guildID];
+    if (!gameSession) {
+        return;
+    }
+
     if (checkBotIsAlone(guildID)) {
-        const gameSession = state.gameSessions[guildID];
-        if (gameSession) {
-            gameSession.endSession();
-        }
+        gameSession.endSession();
+        return;
+    }
+
+    if (!gameSession.finished) {
```

Comment:
Do we need this in voiceChannelSwitch as well?

Reply:
We do, thanks